### ISSUE TITLE
Vite related issue

### ISSUE THREAD

**Issue by wobsoriano** [papyrs/stylo#24]
### Reproduction

Steps to reproduce the issue:

1. Create any [vite](https://vitejs.dev/) app
2.  Install and import library `import '@papyrs/stylo'`

Error in terminal:

```
381|  var ce = (e2) => {
382|    const t2 = e2.h.replace(/-/g, "_"), n2 = e2.q, l2 = ie.get(n2);
383|    return l2 ? l2[t2] : import(`./${n2}.entry.js`).then((e3) => (ie.set(n2, e3), e3[t2]), oe);
   |                                ^
384|  };
385|  var re = /* @__PURE__ */ new Map();
The above dynamic import cannot be analyzed by vite.
See https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations for supported dynamic import formats.
If this is intended to be left as-is, you can use the /* @vite-ignore */ comment inside the import() call to suppress this warning.

  Plugin: vite:import-analysis
  File: /test-project/node_modules/.vite/@papyrs_stylo.js?v=ac2d9e89
```

Error in browser console:

<img width="678" alt="Screen Shot 2022-02-10 at 10 41 16 AM" src="https://user-images.githubusercontent.com/13049130/153326993-de28c3b8-51f6-4846-b956-220a03cfe8d7.png">

**Comment by wobsoriano** [comment]
For now the working solution is to import via `unpkg`

```ts
import "https://unpkg.com/@papyrs/stylo@0.0.4-1/dist/stylo/stylo.esm.js";
```

or

```html
<script type="module" src="https://unpkg.com/@papyrs/stylo@0.0.4-1/dist/stylo/stylo.esm.js"></script>
```

Had to change `stylo@alpha` to `stylo@0.0.4-1` as it's returning 404

**Comment by peterpeterparker** [comment]
I think `import '@papyrs/stylo'` won't work with vite. On the contrary, if I reckon correctly, using the custom elements will - should.

```
import { defineCustomElements } from '@papyrs/stylo/dist/loader';
defineCustomElements();
```

**Comment by wobsoriano** [comment]
Same error
<img width="717" alt="Screen Shot 2022-02-10 at 2 56 31 PM" src="https://user-images.githubusercontent.com/13049130/153353727-e373a3cc-2bb8-4e72-a171-cd99ab78d47a.png">

<img width="616" alt="Screen Shot 2022-02-10 at 2 56 40 PM" src="https://user-images.githubusercontent.com/13049130/153353750-68f6afde-e3a4-4b04-a504-6fc36ea3d21b.png">



> I think `import '@papyrs/stylo'` won't work with vite. On the contrary, if I reckon correctly, using the custom elements will - should.
> 
> ```
> import { defineCustomElements } from '@papyrs/stylo/dist/loader';
> defineCustomElements();
> ```

**Comment by peterpeterparker** [comment]
I needed a coffee, I think it works as following:

```
import { defineCustomElements } from '@papyrs/stylo/dist/components/stylo-editor';
defineCustomElements();
```

However doing so it won't work without installing in addition `@stencil/core` as a dev dependency.

There are several issues open at Ionic and Stencil about it. Until these are solved there is no cleaner way to import it in vite projects I think.

https://github.com/ionic-team/ionic-framework/issues/23823

https://github.com/ionic-team/stencil/issues/2826

https://github.com/ionic-team/stencil/pull/2959

**Comment by peterpeterparker** [comment]
In a svelteki app I wasn't able to load the component as displayed above. As a workaround, I did as following:

```
onMount(async () => {
    const {defineCustomElement} = await import('@papyrs/stylo/dist/components/stylo-editor');
    defineCustomElement();
  });
```

**Comment by peterpeterparker** [comment]
Another tricks, if you are looking to import plugins, these can be imported as following:

```
import {h1, h2, h3, ul} from '@papyrs/stylo/dist/stylo/index.esm';
```

**Comment by peterpeterparker** [comment]
also could load another component as following:

```
const {DeckgoDoc} = await import('@deckdeckgo/core/dist/components/deckgo-doc');
    customElements.get('deckgo-doc') || customElements.define('deckgo-doc', DeckgoDoc);
    ```

**Comment by peterpeterparker** [comment]
in call case I needed `@stencil/core` as a dependency and super important with the same version because there is a breaking change in `v2.9.0` https://github.com/ionic-team/ionic-framework/issues/24153

**Comment by peterpeterparker** [comment]
This has been solved thanks to an improvement in StencilJS and released in `v0.0.38`

See https://github.com/papyrs/stylo/releases/tag/v0.0.38
